Makefile

```diff
@@ -52,7 +59,12 @@ check:
 vendor: check
 	# download the dependencies
 	git submodule update --init
-	$(MAKE) -C vendor all
+	# create+activate a Python venv if not already active
+	if [ -z "$(VIRTUAL_ENV)" ]; then \
+	$(PYTHON) -m venv $(SHAREDIR)/venv && \
+	. $(SHAREDIR)/venv/bin/activate && \
+	pip install -U pip; \
+	fi && $(MAKE) -C vendor all
```

Comment:
@bertsky, @kba, these lines break `make all` in a default build (`VIRTUAL_ENV` not set, `SHAREDIR=/usr/local/share`).

Making a virtual Python environment will fail because of missing write permission.

Using `sudo` will make things even worse because of git conflicts with different users, and it will also create files owned by root in the user's file tree.